hedera-mirror-rosetta/app/services/network/network_service_test.go

```diff
@@ -0,0 +1,227 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ‍
+ */
+
+package network
+
+import (
+	"github.com/coinbase/rosetta-sdk-go/server"
+	rTypes "github.com/coinbase/rosetta-sdk-go/types"
+	"github.com/hashgraph/hedera-mirror-node/hedera-mirror-rosetta/app/domain/types"
+	"github.com/hashgraph/hedera-mirror-node/hedera-mirror-rosetta/app/errors"
+	"github.com/hashgraph/hedera-mirror-node/hedera-mirror-rosetta/app/services/base"
+	"github.com/hashgraph/hedera-mirror-node/hedera-mirror-rosetta/tests/mocks/repository"
+	"github.com/hashgraph/hedera-mirror-node/hedera-mirror-rosetta/tools/maphelper"
+	"github.com/stretchr/testify/assert"
+	"testing"
+)
+
+func getSubject() server.NetworkAPIServicer {
+	baseService := base.NewBaseService(repository.MBlockRepository, repository.MTransactionRepository)
+	return networkAPIService(baseService)
+}
+
+func dummyGenesisBlock() *types.Block {
+	return &types.Block{
+		Index:               1,
+		Hash:                "0x123jsjs",
+		ConsensusStartNanos: 1000000,
+		ConsensusEndNanos:   20000000,
+		ParentIndex:         0,
+		ParentHash:          "",
+	}
+}
+
+func dummyLatestBlock() *types.Block {
+	return &types.Block{
+		Index:               2,
+		Hash:                "0x1323jsjs",
+		ConsensusStartNanos: 40000000,
+		ConsensusEndNanos:   70000000,
+		ParentIndex:         1,
+		ParentHash:          "0x123jsjs",
+	}
+}
+
+func networkAPIService(base base.BaseService) server.NetworkAPIServicer {
+	return NewNetworkAPIService(
+		base,
+		repository.MAddressBookEntryRepository,
+		&rTypes.NetworkIdentifier{
+			Blockchain: "SomeBlockchain",
+			Network:    "SomeNetwork",
+			SubNetworkIdentifier: &rTypes.SubNetworkIdentifier{
+				Network:  "SomeSubNetwork",
+				Metadata: nil,
+			},
+		},
+		&rTypes.Version{
+			RosettaVersion:    "1",
+			NodeVersion:       "1",
+			MiddlewareVersion: nil,
+			Metadata:          nil,
+		},
+	)
+}
+
+func TestNewNetworkAPIService(t *testing.T) {
+	repository.Setup()
+	baseService := base.NewBaseService(repository.MBlockRepository, repository.MTransactionRepository)
+	networkService := networkAPIService(baseService)
+
+	assert.IsType(t, &NetworkAPIService{}, networkService)
+}
+
+func TestNetworkList(t *testing.T) {
+	// given:
+	expectedResult := &rTypes.NetworkListResponse{
+		NetworkIdentifiers: []*rTypes.NetworkIdentifier{
+			{
+				Blockchain: "SomeBlockchain",
+				Network:    "SomeNetwork",
+				SubNetworkIdentifier: &rTypes.SubNetworkIdentifier{
+					Network:  "SomeSubNetwork",
+					Metadata: nil,
+				},
+			},
+		},
+	}
+
+	repository.Setup()
+
+	// when:
+	res, e := getSubject().NetworkList(nil, nil)
+
+	// then:
+	assert.Equal(t, expectedResult, res)
+	assert.Nil(t, e)
+}
+
+func TestNetworkOptions(t *testing.T) {
```

Comment:
I think that you should add 2 tests where:
-  `TypesAsArray` returns error
- `Statuses` returns error
as they are not covered at the moment